cpp/arcticdb/util/configs_map.hpp

```diff
@@ -21,32 +21,32 @@ using namespace arcticdb::proto::config;
 
 class ConfigsMap {
 public:
-    static std::shared_ptr<ConfigsMap> instance_;
-    static std::once_flag init_flag_;
-
     static void init();
-    static std::shared_ptr<ConfigsMap> instance();
-
-    ConfigsMap() :
-        mutex_(std::make_unique<std::mutex>()) {
+    static std::shared_ptr<ConfigsMap>& instance() { 
+        static auto instance_ = std::make_shared<ConfigsMap>();
+        return instance_;
     }
 
 #define HANDLE_TYPE(LABEL, TYPE)     \
     void set_##LABEL(const std::string& label, TYPE val) { \
+        std::lock_guard<std::mutex> lock(mutex_); \
         map_of_##LABEL[boost::to_upper_copy<std::string>(label)] = val; \
     } \
 \
     TYPE get_##LABEL(const std::string& label, TYPE default_val) const { \
+        std::lock_guard<std::mutex> lock(mutex_); \
```

Comment:
Why do we need this guard? We call these `get_` methods all the time, often in fairly performance sensitive blocks (which perhaps we shouldn't), so I'm a bit concerned about the performance impact